### ISSUE TITLE
TraktForVLC 2.0.0a2.x not working ("core warning: connection timed out"?). VLC 2.2.6. Windows 8.1 Pro 64. 

### ISSUE THREAD

**Issue by longbaugh01** [XaF/TraktForVLC#107]
Hi Xaf and anyone else. I've been struggling with this issue for a week and hope I can get some help.

As said in the title, basically, I can't get TraktForVLC to work at all. I've tried multiple of the latest dev releases including the original 2.0.0a2 and up to the latest release TraktForVLC_2.0.0a2.dev14.g31faf4b_windows_x86.exe (Using VLC 2.2.6 32-bit/x86).

I will attach a portion of one log file and then a full one, but let me say that I have inspected the debug messages many many times for different files of both TV and movies, and while there has been some variation, the one thing I feel I keep seeing over and over again is some variation of:
```
core debug: net: connecting to localhost port 1984
core warning: connection timed out
core debug: connection succeeded (socket = 2368(*this number changes))
```

But I have no idea if this is actually part of the issue or not. Running Netstat in Administrator Command Prompt I usually see 1 or more multiple listings of:
```
" Proto  Local Address          Foreign Address        State
 TCP    127.0.0.1:1984         MULTIVAC:64864         TIME_WAIT"
```
While VLC and the media file are running.

I also usually see in every log some variation of:
```
lua debug: Reading buffer; content = Exit: 0 
lua debug: Reading buffer; content = {} 
lua debug: Received data before parsing = Exit: 0 {} 
lua debug: Parsed EXIT_CODE = 0
lua debug: Parsed RESPONSE = {} 
lua debug: (call_helper) Received response: {} 
lua info: No update found for TraktForVLC.
lua debug: Timers ran: { ["@trakt.lua:2365"] = true, ["@trakt.lua:1573"] = true, ["@trakt.lua:2617"] = true, }
```

Again, I don't know how pertinent to the issue these are.

Some steps I have taken (not necessarily in order and not always just once):
-Uninstall/Reinstall TraktForVLC and try different releases going back to original 2.0.0.a2.
-Revoke Access on Trakt, and then re-run authorization using "init_trakt_auth".
-Make sure both incoming and outgoing for VLC is unblocked in Windows Firewall.
-Change port in config .json file to the default one I see in VLC's Lua interface settings which is 4212.

None of these steps have been successful.

Also, I will note that there has been some strange behavior:
-The first time I tried to install TFVlc, everything seemed to go as it should. The VLC player popped up and immediately gave me the Auth Code which worked. BUT, every time after this I have re-run the auth, or completely reinstalled TFVlc, the VLC player pops up and runs the logo video repeatedly without giving me a code until I pause it. And even before I figured out I had to pause it, it wouldn't give me a code so I'd close it, but then the next media file I tried would pause and give me a code, sometimes it would do this many, many times in a row when playing same or different media files.

-I use a VLC remote Android App which uses the VLC "web" interface option, port 8080, and requires me to use a password I put into the LUA http password section in VLC's settings. Ever since I started trying to use TFVlc, this remote no longer works. I know this isn't your problem, but is TFVLC putting in some sort of password into the lua http password field in settings that isn't specified in the README or config?

I will now get to the log files. This first is just a portion of a log file because it's so long (I have also redacted some information such as filenames):

```
core debug: auto hiding mouse cursor
core warning: connection timed out
core debug: connection succeeded (socket = 608)
lua debug: Reading buffer; content = Exit: 0 
lua debug: Reading buffer; content = { "date": "1525309746.859000", "format": 

"%s.%f", "timezone": "UTC" } 
lua debug: Received data before parsing = Exit: 0 { "date": 

"1525309746.859000", "format": "%s.%f", "timezone": "UTC" } 
lua debug: Parsed EXIT_CODE = 0
lua debug: Parsed RESPONSE = { "date": "1525309746.859000", "format": "%s.%f", 

"timezone": "UTC" } 
lua debug: (call_helper) Received response: { "date": "1525309746.859000", 

"format": "%s.%f", "timezone": "UTC" } 
lua debug: (call_helper) Executing command: "update" "--vlc-lua-directory" 

"C:\Program Files (x86)\VideoLAN\VLC\lua" "--vlc-config" "C:\Users

\longbaugh01\AppData\Roaming\vlc" "--yes" "--release-type" "stable" "--

service" "--service-host" "localhost" "--service-port" "1984" "--install" "--

discard-install-output"
core debug: net: connecting to localhost port 1984
core debug: auto hiding mouse cursor
core debug: auto hiding mouse cursor
core warning: connection timed out
core debug: connection succeeded (socket = 2368)
lua debug: Reading buffer; content = Exit: 0 
lua debug: Reading buffer; content = {} 
lua debug: Received data before parsing = Exit: 0 {} 
lua debug: Parsed EXIT_CODE = 0
lua debug: Parsed RESPONSE = {} 
lua debug: (call_helper) Received response: {} 
lua info: No update found for TraktForVLC.
lua debug: Timers ran: { ["@trakt.lua:2299"] = true, ["@trakt.lua:1521"] = 

true, ["@trakt.lua:2545"] = true, }
lua debug: (call_helper) Executing command: "--quiet" "resolve" "--meta" 

"{\"_STATISTICS_WRITING_APP\":\"mkvmerge v9.6.0 ('Slave To Your Mind') 64bit

\",\"NUMBER_OF_FRAMES\":\"810\",\"NUMBER_OF_BYTES\":\"39763\",\"BPS\":\"92\",

\"title\":\"The Keepers S01E02\",\"_STATISTICS_TAGS\":\"BPS DURATION 

NUMBER_OF_FRAMES NUMBER_OF_BYTES\",\"DURATION\":\"00:57:42.230000000\",

\"filename\":\"The.Keepers.S01E02.mkv\",\"ENCODER\":\"Lavc57.86.103 ssa\",

\"_STATISTICS_WRITING_DATE_UTC\":\"2017-05-19 16:07:55\"}" "--duration" 

"3550.133" "--hash" "f58a3daa1ffe5ea5" "--size" "661013381"
core debug: net: connecting to localhost port 1984
core debug: auto hiding mouse cursor
core warning: connection timed out
core debug: connection succeeded (socket = 2368)
lua debug: Reading buffer; content = Exit: 0 
lua debug: Reading buffer; content = [ { "@type": 

"imdb.api.title.auxiliary.v1", "base": { "disambiguation": "I", "episode": 2, 

"id": "/title/tt6510708/", "nextEpisode": "/title/tt6510744/", "parentTitle": 

{ "disambiguation": "I", "id": "/title/tt6502890/", "image": { "height": 791, 

"id": "/title/tt6502890/images/rm3384359936", "url": "https://ia.media-

imdb.com/images/M/MV5BMGFjYWNmMGYtZWNiMC00YmIwLThjZTUtN2QxM2M3ZTQxNTg2L2ltYWdl

XkEyXkFqcGdeQXVyMjU5NDg0MjY@._V1_.jpg", "width": 612 }, "title": "The 

Keepers", "titleType": "tvSeries", "year": 2017 }, "previousEpisode": 

"/title/tt6510692/", "season": 1, "seriesStartYear": 2017, "title": "Dream 

World", "titleType": "tvEpisode", "tmdbid": 1318548, "tvdbid": 6066787, 

"year": 2017 }, "metacriticScore": { "@type": 

"imdb.api.title.metacritic.score", "reviewCount": 0, "userRatingCount": 0 }, 

"ratings": { "canRate": true, "disambiguation": "I", "episode": 2, "id": 

"/title/tt6510708/", "season": 1, "title": "Dream World", 
lua debug: Reading buffer; content = "titleType": "tvEpisode", "year": 2017 } 

} ] 
lua debug: Received data before parsing = Exit: 0 [ { "@type": 

"imdb.api.title.auxiliary.v1", "base": { "disambiguation": "I", "episode": 2, 

"id": "/title/tt6510708/", "nextEpisode": "/title/tt6510744/", "parentTitle": 

{ "disambiguation": "I", "id": "/title/tt6502890/", "image": { "height": 791, 

"id": "/title/tt6502890/images/rm3384359936", "url": "https://ia.media-

imdb.com/images/M/MV5BMGFjYWNmMGYtZWNiMC00YmIwLThjZTUtN2QxM2M3ZTQxNTg2L2ltYWdl

XkEyXkFqcGdeQXVyMjU5NDg0MjY@._V1_.jpg", "width": 612 }, "title": "The 

Keepers", "titleType": "tvSeries", "year": 2017 }, "previousEpisode": 

"/title/tt6510692/", "season": 1, "seriesStartYear": 2017, "title": "Dream 

World", "titleType": "tvEpisode", "tmdbid": 1318548, "tvdbid": 6066787, 

"year": 2017 }, "metacriticScore": { "@type": 

"imdb.api.title.metacritic.score", "reviewCount": 0, "userRatingCount": 0 }, 

"ratings": { "canRate": true, "disambiguation": "I", "episode": 2, "id": 

"/title/tt6510708/", "season": 1, "title": "Dream World", "titleType": 

"tvEpisode", "year": 2017 } } ] 
lua debug: Parsed EXIT_CODE = 0
lua debug: Parsed RESPONSE = [ { "@type": "imdb.api.title.auxiliary.v1", 

"base": { "disambiguation": "I", "episode": 2, "id": "/title/tt6510708/", 

"nextEpisode": "/title/tt6510744/", "parentTitle": { "disambiguation": "I", 

"id": "/title/tt6502890/", "image": { "height": 791, "id": 

"/title/tt6502890/images/rm3384359936", "url": "https://ia.media-

imdb.com/images/M/MV5BMGFjYWNmMGYtZWNiMC00YmIwLThjZTUtN2QxM2M3ZTQxNTg2L2ltYWdl

XkEyXkFqcGdeQXVyMjU5NDg0MjY@._V1_.jpg", "width": 612 }, "title": "The 

Keepers", "titleType": "tvSeries", "year": 2017 }, "previousEpisode": 

"/title/tt6510692/", "season": 1, "seriesStartYear": 2017, "title": "Dream 

World", "titleType": "tvEpisode", "tmdbid": 1318548, "tvdbid": 6066787, 

"year": 2017 }, "metacriticScore": { "@type": 

"imdb.api.title.metacritic.score", "reviewCount": 0, "userRatingCount": 0 }, 

"ratings": { "canRate": true, "disambiguation": "I", "episode": 2, "id": 

"/title/tt6510708/", "season": 1, "title": "Dream World", "titleType": 

"tvEpisode", "year": 2017 } } ] 
lua debug: (call_helper) Received response: [ { "@type": 

"imdb.api.title.auxiliary.v1", "base": { "disambiguation": "I", "episode": 2, 

"id": "/title/tt6510708/", "nextEpisode": "/title/tt6510744/", "parentTitle": 

{ "disambiguation": "I", "id": "/title/tt6502890/", "image": { "height": 791, 

"id": "/title/tt6502890/images/rm3384359936", "url": "https://ia.media-

imdb.com/images/M/MV5BMGFjYWNmMGYtZWNiMC00YmIwLThjZTUtN2QxM2M3ZTQxNTg2L2ltYWdl

XkEyXkFqcGdeQXVyMjU5NDg0MjY@._V1_.jpg", "width": 612 }, "title": "The 

Keepers", "titleType": "tvSeries", "year": 2017 }, "previousEpisode": 

"/title/tt6510692/", "season": 1, "seriesStartYear": 2017, "title": "Dream 

World", "titleType": "tvEpisode", "tmdbid": 1318548, "tvdbid": 6066787, 

"year": 2017 }, "metacriticScore": { "@type": 

"imdb.api.title.metacritic.score", "reviewCount": 0, "userRatingCount": 0 }, 

"ratings": { "canRate": true, "disambiguation": "I", "episode": 2, "id": 

"/title/tt6510708/", "season": 1, "title": "Dream World", "titleType": 

"tvEpisode", "year": 2017 } } ] 
lua debug: (call_helper) Executing command: "date" "--format" "%s.%f"
core debug: net: connecting to localhost port 1984
core warning: connection timed out
core debug: connection succeeded (socket = 2368)
lua debug: Reading buffer; content = Exit: 0 
lua debug: Reading buffer; content = { "date": "1525309768.454000", "format": 

"%s.%f", "timezone": "UTC" } 
lua debug: Received data before parsing = Exit: 0 { "date": 

"1525309768.454000", "format": "%s.%f", "timezone": "UTC" } 
lua debug: Parsed EXIT_CODE = 0
lua debug: Parsed RESPONSE = { "date": "1525309768.454000", "format": "%s.%f", 

"timezone": "UTC" } 
lua debug: (call_helper) Received response: { "date": "1525309768.454000", 

"format": "%s.%f", "timezone": "UTC" } 
lua info: Force saving cache
lua debug: Writing to C:\Users\longbaugh01\AppData\Roaming\vlc

\trakt_cache.json: "{ "file:*REDACTED*#2571.402":{ "last_use":1524915404.26, 

"imdb":[{ "base":{ "seriesStartYear":2018, "episode":5, 

"id":"/title/tt6687658/", "parentTitle":{ "titleType":"tvSeries", "image":{ 

"id":"/title/tt2708480/images/rm121459968", "url":"https://ia.media-

imdb.com/images/M/MV5BMTAyMjEzMzg0NTleQTJeQWpwZ15BbWU4MDY5NDE4OTQz._V1_.jpg", 

"height":979, "width":690 }, "title":"The Terror", "id":"/title/tt2708480/", 

"year":2018 }, "nextEpisode":"/title/tt6687664/", "titleType":"tvEpisode", 

"image":{ "id":"/title/tt6687658/images/rm1347832064", 

"url":"https://ia.media-

imdb.com/images/M/MV5BYTFlZTM2NTAtOGUxOC00NjhjLTk2NGQtYjI4NmFmY2NkYmVjXkEyXkFq

cGdeQXVyNDg4MjkzNDk@._V1_.jpg", "height":1080, "width":1920 }, "title":"First 

Shot a Winner, Lads", "season":1, "previousEpisode":"/title/tt6182174/", 

"tmdbid":1464300, "year":2018 } }], "meta":{ "showName":"The Terror", 

"title":"The Terror S01E05", "filename":"*REDACTED*", "episodeNumber":"05", 

"seasonNumber":"01" }, "duration":2571.402, "uri_path":"*REDACTED*", 

"uri_proto":"file", "imdb_details":{ "play_total":2571.402, 

"missing_time_duration":2571.402, "per_media":[{ "type":"episode", "to":{ 

"vlctime":2571.402, "time":2571.402, "percent":1 }, "from":{ "vlctime":0, 

"time":0, "percent":0 }, "vlcduration":2571.402, "duration":2571.402, 

"name":"The Terror (2018) - S01E05 - First Shot a Winner, Lads" }], 

"play_factor":1 }, "hash":"a623e41d8a06d5c4", "size":1145849501 }, }
lua info: The Keepers (2017) - S01E02 - Dream World is paused! :| (6.39e-

007/3550.133)
lua debug: Timers ran: { ["@trakt.lua:2299"] = false, ["@trakt.lua:1521"] = 

false, ["@trakt.lua:2545"] = false, }
lua debug: (call_helper) Executing command: "date" "--format" "%s.%f"
core debug: net: connecting to localhost port 1984
core warning: connection timed out
core debug: connection succeeded (socket = 2380)
lua debug: Reading buffer; content = Exit: 0 
lua debug: Reading buffer; content = { "date": "1525309774.471000", "format": 

"%s.%f", "timezone": "UTC" } 
lua debug: Received data before parsing = Exit: 0 { "date": 

"1525309774.471000", "format": "%s.%f", "timezone": "UTC" } 
lua debug: Parsed EXIT_CODE = 0
lua debug: Parsed RESPONSE = { "date": "1525309774.471000", "format": "%s.%f", 

"timezone": "UTC" } 
lua debug: (call_helper) Received response: { "date": "1525309774.471000", 

"format": "%s.%f", "timezone": "UTC" } 
lua info: The Keepers (2017) - S01E02 - Dream World is paused! :| (6.39e-

007/3550.133)
lua debug: Timers ran: { ["@trakt.lua:2299"] = false, ["@trakt.lua:1521"] = 

false, ["@trakt.lua:2545"] = false, }
lua debug: (call_helper) Executing command: "date" "--format" "%s.%f"
core debug: net: connecting to localhost port 1984
core warning: connection timed out
core debug: connection succeeded (socket = 2380)
lua debug: Reading buffer; content = Exit: 0 
lua debug: Reading buffer; content = { "date": "1525309780.485000", "format": 

"%s.%f", "timezone": "UTC" } 
lua debug: Received data before parsing = Exit: 0 { "date": 

"1525309780.485000", "format": "%s.%f", "timezone": "UTC" } 
lua debug: Parsed EXIT_CODE = 0
lua debug: Parsed RESPONSE = { "date": "1525309780.485000", "format": "%s.%f", 

"timezone": "UTC" } 
lua debug: (call_helper) Received response: { "date": "1525309780.485000", 

"format": "%s.%f", "timezone": "UTC" } 
lua info: The Keepers (2017) - S01E02 - Dream World is paused! :| (6.39e-

007/3550.133)
lua debug: Timers ran: { ["@trakt.lua:2299"] = false, ["@trakt.lua:1521"] = 

false, ["@trakt.lua:2545"] = false, }
lua debug: (call_helper) Executing command: "date" "--format" "%s.%f"
core debug: net: connecting to localhost port 1984
core warning: connection timed out
core debug: connection succeeded (socket = 2380)
lua debug: Reading buffer; content = Exit: 0 
lua debug: Reading buffer; content = { "date": "1525309786.497000", "format": 

"%s.%f", "timezone": "UTC" } 
lua debug: Received data before parsing = Exit: 0 { "date": 

"1525309786.497000", "format": "%s.%f", "timezone": "UTC" } 
lua debug: Parsed EXIT_CODE = 0
lua debug: Parsed RESPONSE = { "date": "1525309786.497000", "format": "%s.%f", 

"timezone": "UTC" } 
lua debug: (call_helper) Received response: { "date": "1525309786.497000", 

"format": "%s.%f", "timezone": "UTC" } 
lua info: The Keepers (2017) - S01E02 - Dream World is paused! :| (6.39e-

007/3550.133)
lua debug: Timers ran: { ["@trakt.lua:2299"] = false, ["@trakt.lua:1521"] = 

false, ["@trakt.lua:2545"] = false, }
lua debug: (call_helper) Executing command: "date" "--format" "%s.%f"
core debug: net: connecting to localhost port 1984
core warning: connection timed out
core debug: connection succeeded (socket = 2384)
lua debug: Reading buffer; content = Exit: 0 
lua debug: Reading buffer; content = { "date": "1525309792.511000", "format": 

"%s.%f", "timezone": "UTC" } 
lua debug: Received data before parsing = Exit: 0 { "date": 

"1525309792.511000", "format": "%s.%f", "timezone": "UTC" } 
lua debug: Parsed EXIT_CODE = 0
lua debug: Parsed RESPONSE = { "date": "1525309792.511000", "format": "%s.%f", 

"timezone": "UTC" } 
lua debug: (call_helper) Received response: { "date": "1525309792.511000", 

"format": "%s.%f", "timezone": "UTC" } 
lua info: The Keepers (2017) - S01E02 - Dream World is paused! :| (6.39e-

007/3550.133)
lua debug: Timers ran: { ["@trakt.lua:2299"] = false, ["@trakt.lua:1521"] = 

false, ["@trakt.lua:2545"] = false, }
lua debug: (call_helper) Executing command: "date" "--format" "%s.%f"
core debug: net: connecting to localhost port 1984
core warning: connection timed out
core debug: connection succeeded (socket = 2384)
lua debug: Reading buffer; content = Exit: 0 
lua debug: Reading buffer; content = { "date": "1525309798.525000", "format": 

"%s.%f", "timezone": "UTC" } 
lua debug: Received data before parsing = Exit: 0 { "date": 

"1525309798.525000", "format": "%s.%f", "timezone": "UTC" } 
lua debug: Parsed EXIT_CODE = 0
lua debug: Parsed RESPONSE = { "date": "1525309798.525000", "format": "%s.%f", 

"timezone": "UTC" } 
lua debug: (call_helper) Received response: { "date": "1525309798.525000", 

"format": "%s.%f", "timezone": "UTC" } 
lua debug: Writing to C:\Users\longbaugh01\AppData\Roaming\vlc

\trakt_cache.json:
```

This second log file is the latest using latest version of TFVLC. I played a TV episode (different from first log file episode) and let it play for 30 seconds, skipped around and to the end and played the last 30 minutes and then let it finish:

```
core debug: VLC media player - 2.2.6 Umbrella
core debug: Copyright © 1996-2017 the VideoLAN team
core debug: revision 2.2.6-0-g1aae789
core debug: configured with ../extras/package/win32/../../../configure '--enable-update-check' '--enable-lua' '--enable-faad' '--enable-flac' '--enable-theora' '--enable-twolame' '--enable-quicktime' '--enable-avcodec' '--enable-merge-ffmpeg' '--enable-dca' '--enable-mpc' '--enable-libass' '--enable-x264' '--enable-schroedinger' '--enable-realrtsp' '--enable-live555' '--enable-dvdread' '--enable-shout' '--enable-goom' '--enable-caca' '--disable-sdl' '--enable-qt' '--enable-skins2' '--enable-sse' '--enable-mmx' '--enable-libcddb' '--enable-zvbi' '--disable-telx' '--enable-nls' '--host=i686-w64-mingw32' 'host_alias=i686-w64-mingw32'
core debug: auto hiding mouse cursor
core debug: VoutDisplayEvent 'mouse button' 0 t=8
core debug: VoutDisplayEvent 'mouse button' 0 t=9
clock error: Timestamp conversion failed (delay 6000000, buffering 100000, bound 3000000)
core error: Could not convert timestamp 5325873323999
core debug: discarded audio buffer
core warning: connection timed out
core debug: connection succeeded (socket = 736)
lua debug: Reading buffer; content = Exit: 0 
lua debug: Reading buffer; content = {} 
lua debug: Received data before parsing = Exit: 0 {} 
lua debug: Parsed EXIT_CODE = 0
lua debug: Parsed RESPONSE = {} 
lua debug: (call_helper) Received response: {} 
lua info: No update found for TraktForVLC.
lua debug: Timers ran: { ["@trakt.lua:2365"] = true, ["@trakt.lua:1573"] = true, ["@trakt.lua:2617"] = true, }
core debug: auto hiding mouse cursor
core debug: auto hiding mouse cursor
core debug: auto hiding mouse cursor
core debug: auto hiding mouse cursor
core debug: auto hiding mouse cursor
core debug: auto hiding mouse cursor
core debug: auto hiding mouse cursor
core debug: auto hiding mouse cursor
core debug: VoutDisplayEvent 'mouse button' 0 t=8
core debug: VoutDisplayEvent 'mouse button' 0 t=9
core debug: VoutDisplayEvent 'double click'
qt4 debug: Qt: Fullscreen state changed
qt4 debug: Qt: Entering Fullscreen
core debug: VoutDisplayEvent 'resize' 1360x768 fullscreen
core debug: VoutDisplayEvent 'mouse button' 0 t=8
core debug: VoutDisplayEvent 'mouse button' 0 t=9
core debug: VoutDisplayEvent 'double click'
qt4 debug: Qt: Fullscreen state changed
qt4 debug: Qt: Quitting Fullscreen
core debug: VoutDisplayEvent 'resize' 1360x747 fullscreen
core debug: VoutDisplayEvent 'resize' 1344x656 fullscreen
core debug: VoutDisplayEvent 'resize' 1280x720 fullscreen
core debug: VoutDisplayEvent 'mouse button' 0 t=8
core debug: VoutDisplayEvent 'mouse button' 0 t=9
core debug: VoutDisplayEvent 'double click'
qt4 debug: Qt: Fullscreen state changed
qt4 debug: Qt: Entering Fullscreen
core debug: VoutDisplayEvent 'resize' 1360x768 fullscreen
core debug: auto hiding mouse cursor
core warning: early picture skipped
core debug: discarded audio buffer
mkv debug: seek request to -1 (0.476000%)
mkv debug: seek got 1605792000 - 271550072
core debug: Buffering 0%
core debug: Buffering 0%
core debug: Buffering 3%
core debug: Buffering 7%
core debug: Buffering 10%
core debug: Buffering 14%
core debug: Buffering 17%
core debug: Buffering 21%
core debug: Buffering 24%
core debug: Buffering 28%
core debug: Buffering 31%
core debug: Buffering 35%
core debug: Buffering 38%
core debug: Buffering 42%
core debug: Buffering 45%
core debug: Buffering 49%
core debug: Buffering 52%
core debug: Buffering 56%
core debug: Buffering 59%
core debug: Buffering 63%
core debug: Buffering 66%
core debug: Buffering 70%
core debug: Buffering 73%
core debug: End of audio preroll
core debug: Buffering 77%
core debug: Buffering 80%
core debug: Buffering 84%
core debug: Buffering 87%
core debug: Buffering 91%
core debug: Buffering 94%
core debug: Buffering 98%
core debug: Stream buffering done (9258 ms in 263 ms)
core debug: auto hiding mouse cursor
core debug: VoutDisplayEvent 'mouse button' 0 t=8
core debug: VoutDisplayEvent 'mouse button' 0 t=9
core debug: VoutDisplayEvent 'double click'
qt4 debug: Qt: Fullscreen state changed
qt4 debug: Qt: Quitting Fullscreen
core debug: VoutDisplayEvent 'resize' 1344x656 fullscreen
core debug: VoutDisplayEvent 'resize' 1280x720 fullscreen
core debug: End of video preroll
core debug: Received first picture
core debug: Decoder wait done in 581 ms
core debug: picture might be displayed late (missing 2 ms)
core debug: auto hiding mouse cursor
core debug: auto hiding mouse cursor
core debug: auto hiding mouse cursor
core debug: auto hiding mouse cursor
core debug: auto hiding mouse cursor
core debug: auto hiding mouse cursor
core debug: auto hiding mouse cursor
core debug: auto hiding mouse cursor
core debug: auto hiding mouse cursor
core debug: auto hiding mouse cursor
core debug: VoutDisplayEvent 'mouse button' 0 t=8
core debug: VoutDisplayEvent 'mouse button' 0 t=9
core debug: VoutDisplayEvent 'mouse button' 0 t=8
core debug: VoutDisplayEvent 'mouse button' 0 t=9
core debug: VoutDisplayEvent 'double click'
qt4 debug: Qt: Fullscreen state changed
qt4 debug: Qt: Entering Fullscreen
core debug: VoutDisplayEvent 'resize' 1360x768 fullscreen
core debug: auto hiding mouse cursor
core warning: early picture skipped
core debug: discarded audio buffer
mkv debug: seek request to -1 (0.968000%)
mkv debug: seek got 3263698000 - 620179452
core debug: Buffering 0%
core debug: Buffering 0%
core debug: Buffering 2%
core debug: Buffering 4%
core debug: Buffering 6%
core debug: Buffering 8%
core debug: Buffering 11%
core debug: Buffering 13%
core debug: Buffering 15%
core debug: Buffering 17%
core debug: Buffering 20%
core debug: Buffering 22%
core debug: Buffering 24%
core debug: Buffering 26%
core debug: Buffering 29%
core debug: Buffering 31%
core debug: Buffering 33%
core debug: Buffering 35%
core debug: Buffering 38%
core debug: Buffering 40%
core debug: Buffering 42%
core debug: Buffering 44%
core debug: Buffering 47%
core debug: Buffering 49%
core debug: Buffering 51%
core debug: Buffering 53%
core debug: Buffering 56%
core debug: Buffering 58%
core debug: Buffering 60%
core debug: Buffering 62%
core debug: Buffering 65%
core debug: Buffering 67%
core debug: Buffering 69%
core debug: Buffering 72%
core debug: Buffering 74%
core debug: Buffering 76%
core debug: Buffering 78%
core debug: Buffering 80%
core debug: Buffering 83%
core debug: Buffering 85%
core debug: Buffering 87%
core debug: Buffering 90%
core debug: Buffering 92%
core debug: Buffering 94%
core debug: Buffering 96%
core debug: Buffering 99%
core debug: Stream buffering done (14347 ms in 182 ms)
core debug: End of audio preroll
core debug: End of video preroll
core debug: Received first picture
core debug: Decoder wait done in 2047 ms
core warning: early picture skipped
core debug: discarded audio buffer
mkv debug: seek request to -1 (0.982000%)
mkv debug: seek got 3315834000 - 626682052
core debug: Buffering 0%
core debug: Buffering 0%
core debug: Buffering 3%
core debug: Buffering 6%
core debug: Buffering 10%
core debug: Buffering 13%
core debug: Buffering 17%
core debug: Buffering 20%
core debug: Buffering 24%
core debug: Buffering 27%
core debug: Buffering 30%
core debug: Buffering 34%
core debug: Buffering 37%
core debug: Buffering 41%
core debug: Buffering 44%
core debug: Buffering 47%
core debug: Buffering 51%
core debug: Buffering 54%
core debug: Buffering 58%
core debug: Buffering 61%
core debug: Buffering 65%
core debug: Buffering 68%
core debug: Buffering 71%
core debug: Buffering 75%
core debug: Buffering 78%
core debug: Buffering 82%
core debug: Buffering 85%
core debug: Buffering 89%
core debug: Buffering 92%
core debug: Buffering 95%
core debug: Buffering 99%
core debug: Stream buffering done (9591 ms in 88 ms)
core debug: End of audio preroll
core debug: End of video preroll
core debug: Received first picture
core debug: Decoder wait done in 186 ms
core warning: can't get output picture
core warning: can't get output picture
core warning: can't get output picture
core debug: discarded audio buffer
mkv debug: seek request to -1 (0.988000%)
mkv debug: seek got 3336688000 - 627549078
core debug: Buffering 0%
core debug: Buffering 0%
core debug: Buffering 3%
core debug: Buffering 7%
core debug: Buffering 10%
core debug: Buffering 14%
core debug: Buffering 18%
core debug: Buffering 21%
core debug: Buffering 25%
core debug: Buffering 28%
core debug: Buffering 32%
core debug: Buffering 36%
core debug: Buffering 39%
core debug: Buffering 43%
core debug: Buffering 47%
core debug: Buffering 50%
core debug: Buffering 54%
core debug: Buffering 58%
core debug: Buffering 61%
core debug: Buffering 65%
core debug: Buffering 69%
core debug: Buffering 72%
core debug: Buffering 76%
core debug: Buffering 80%
core debug: Buffering 83%
core debug: Buffering 87%
core debug: Buffering 90%
core debug: Buffering 94%
core debug: Buffering 98%
core debug: Stream buffering done (8924 ms in 160 ms)
core debug: End of audio preroll
core debug: End of video preroll
core debug: Received first picture
core debug: Decoder wait done in 157 ms
core warning: can't get output picture
core warning: can't get output picture
core warning: can't get output picture
core debug: discarded audio buffer
mkv debug: seek request to -1 (0.992000%)
mkv debug: seek got 3347115000 - 628286390
core debug: Buffering 0%
core debug: Buffering 0%
core debug: Buffering 2%
core debug: Buffering 5%
core debug: Buffering 7%
core debug: Buffering 10%
core debug: Buffering 13%
core debug: Buffering 15%
core debug: Buffering 18%
core debug: Buffering 21%
core debug: Buffering 24%
core debug: Buffering 26%
core debug: Buffering 29%
core debug: Buffering 32%
core debug: Buffering 34%
core debug: Buffering 37%
core debug: Buffering 40%
core debug: Buffering 42%
core debug: Buffering 45%
core debug: Buffering 48%
core debug: Buffering 50%
core debug: Buffering 53%
core debug: Buffering 56%
core debug: Buffering 59%
core debug: Buffering 61%
core debug: Buffering 64%
core debug: Buffering 67%
core debug: Buffering 69%
core debug: Buffering 72%
core debug: Buffering 75%
core debug: Buffering 78%
core debug: Buffering 80%
core debug: Buffering 83%
core debug: Buffering 85%
core debug: Buffering 88%
core debug: Buffering 91%
core debug: Buffering 94%
core debug: Buffering 96%
core debug: Buffering 99%
core debug: Stream buffering done (12117 ms in 213 ms)
core debug: End of audio preroll
core debug: End of video preroll
core debug: Received first picture
core debug: Decoder wait done in 316 ms
core debug: VoutDisplayEvent 'mouse button' 0 t=8
core debug: VoutDisplayEvent 'mouse button' 0 t=9
core debug: VoutDisplayEvent 'double click'
qt4 debug: Qt: Fullscreen state changed
qt4 debug: Qt: Quitting Fullscreen
core debug: VoutDisplayEvent 'resize' 1344x656 fullscreen
core debug: VoutDisplayEvent 'resize' 1280x720 fullscreen
core debug: auto hiding mouse cursor
core debug: auto hiding mouse cursor
core debug: auto hiding mouse cursor
core debug: auto hiding mouse cursor
core debug: auto hiding mouse cursor
mkv warning: find KaxCues FIXME
mkv warning: cannot get block EOF?
core debug: EOF reached
core debug: waiting decoder fifos to empty
core debug: waiting decoder fifos to empty
core debug: waiting decoder fifos to empty
core debug: waiting decoder fifos to empty
core debug: waiting decoder fifos to empty
core debug: waiting decoder fifos to empty
core debug: waiting decoder fifos to empty
core debug: waiting decoder fifos to empty
core debug: waiting decoder fifos to empty
core debug: waiting decoder fifos to empty
core debug: waiting decoder fifos to empty
core debug: waiting decoder fifos to empty
core debug: waiting decoder fifos to empty
core debug: waiting decoder fifos to empty
core debug: waiting decoder fifos to empty
core debug: waiting decoder fifos to empty
core debug: waiting decoder fifos to empty
core debug: waiting decoder fifos to empty
core debug: waiting decoder fifos to empty
core debug: waiting decoder fifos to empty
core debug: waiting decoder fifos to empty
core debug: waiting decoder fifos to empty
core debug: waiting decoder fifos to empty
core debug: waiting decoder fifos to empty
core debug: waiting decoder fifos to empty
core debug: waiting decoder fifos to empty
core debug: waiting decoder fifos to empty
core debug: waiting decoder fifos to empty
core debug: waiting decoder fifos to empty
core debug: waiting decoder fifos to empty
core debug: waiting decoder fifos to empty
core debug: waiting decoder fifos to empty
core debug: waiting decoder fifos to empty
core debug: waiting decoder fifos to empty
core debug: waiting decoder fifos to empty
core debug: waiting decoder fifos to empty
core debug: waiting decoder fifos to empty
core debug: waiting decoder fifos to empty
core debug: waiting decoder fifos to empty
core debug: waiting decoder fifos to empty
core debug: waiting decoder fifos to empty
core debug: waiting decoder fifos to empty
core debug: auto hiding mouse cursor
core debug: waiting decoder fifos to empty
core debug: waiting decoder fifos to empty
core debug: waiting decoder fifos to empty
core debug: waiting decoder fifos to empty
core debug: waiting decoder fifos to empty
core debug: waiting decoder fifos to empty
core debug: waiting decoder fifos to empty
core debug: waiting decoder fifos to empty
core debug: waiting decoder fifos to empty
core debug: waiting decoder fifos to empty
core debug: auto hiding mouse cursor
core debug: waiting decoder fifos to empty
core debug: waiting decoder fifos to empty
core debug: waiting decoder fifos to empty
core debug: waiting decoder fifos to empty
core debug: waiting decoder fifos to empty
core debug: waiting decoder fifos to empty
core debug: waiting decoder fifos to empty
core debug: waiting decoder fifos to empty
core debug: waiting decoder fifos to empty
core debug: waiting decoder fifos to empty
core debug: waiting decoder fifos to empty
core debug: waiting decoder fifos to empty
core debug: waiting decoder fifos to empty
core debug: waiting decoder fifos to empty
core debug: waiting decoder fifos to empty
core debug: waiting decoder fifos to empty
core debug: waiting decoder fifos to empty
core debug: waiting decoder fifos to empty
core debug: waiting decoder fifos to empty
core debug: waiting decoder fifos to empty
core debug: auto hiding mouse cursor
core debug: waiting decoder fifos to empty
core debug: waiting decoder fifos to empty
core debug: waiting decoder fifos to empty
core debug: finished input
core debug: removing module "avcodec"
avcodec debug: ffmpeg codec (MPEG-H Part2/HEVC (H.265)) stopped
core debug: killing decoder fourcc `hevc', 0 PES in FIFO
core debug: saving a free vout
core debug: reusing provided vout
qt4 debug: Qt: Fullscreen state changed
core debug: Filter 07e40534 removed from chain
core debug: removing module "adjust"
core debug: Adding 'adjust' as interactive
core debug: looking for video filter2 module matching "adjust": 58 candidates
core debug: using video filter2 module "adjust"
core debug: Filter 'adjust' (07e40534) appended to chain
core debug: removing module "faad"
core debug: killing decoder fourcc `mp4a', 0 PES in FIFO
core debug: removing module "samplerate"
core debug: removing module "scaletempo"
directsound debug: closing audio device
core debug: removing module "float_mixer"
core debug: keeping audio output
core debug: removing module "libass"
core debug: killing decoder fourcc `ssa ', 0 PES in FIFO
core debug: removing module "mkv"
core debug: Program doesn't contain anymore ES
mkv debug: Stopping the UI Hook
core debug: removing module "record"
core debug: removing module "filesystem"
core debug: dead input
core debug: changing item without a request (current 0/1)
core debug: nothing to play
core debug: destroying useless vout
core debug: removing module "direct3d"
direct3d debug: Direct3D scene released successfully
qt4 debug: IM: Deleting the input
direct3d debug: Win32 Vout EventThread terminating
direct3d debug: Win32VoutCloseWindow
direct3d debug: WinProc WM_DESTROY
core debug: Filter 07e40534 removed from chain
core debug: removing module "adjust"
qt4 debug: releasing video...
qt4 debug: Video is not needed anymore
core debug: removing module "freetype"
core debug: removing module "yuvp"
core debug: removing module "swscale"
```

Thank you for your time in this matter and I really hope someone can help me out. The only reason I even signed up for Trakt was because it had a VLC plugin, spent hours adding episodes I'd watched to Trakt, and if I can't get this to work then Trakt is kind of useless to me and I wasted a lot of time.

Again, thanks.

**Comment by XaF** [comment]
Hi there,

Thanks for all the explanation in your report. However, you didn't say what was and was not working with TraktForVLC. Do you see things appearing on trakt.tv while watching or not at all ?

Now let me reply to a few of your concerns:
 - You don't have to change anything in the lua config, and besides adding `trakt` in the list of modules to load, TraktForVLC shouldn't impact any other plugin
 - Connections to port 1984 is because you're on Windows and the port 1984 is used for the TraktForVLC service in Python that's used for resolution and all (the `trakt_helper`)
 - The message you see at the beginning after that connection is merely TraktForVLC checking for available updates
 - TraktForVLC is still in alpha, so please ideally use the latest releases to keep up with the latest bug fixes (I'm not releasing a new alpha per bugfix)

Regarding your logs:
 - In the first one, everything seems to be in order, the media was resolved
 - In the second one, I don't see any TraktForVLC logs beside the check for update... But those logs are difficult to read as they do not seem to be prefixed by the lua module generating them (so, in this case, `[trakt]`)

**Comment by veselind** [comment]
Hey XaF,

Not sure if my issue is the same as longbaugh01's, but I'm not getting any scrobbling on my side either. Running Windows 10 x64, latest TFV and VLC 3.0.2. I installed TFV using the `--vlc` and `--vlc-lua-directory` arguments, since my VLC isn't in the default install location. My logs are spammed by this:

    [0000000004b47740] [trakt] main interface error: connection failed: Connection refused by peer
    [0000000004b47740] [trakt] lua interface error: (call_helper) Command: "--quiet" "resolve" "--meta" "{\"setting\":\" HAS_INDEX\",\"title\":\"Ed, Edd n' Eddy - S01E20\",\"showName\":\"Ed, Edd n' Eddy -\",\"filename\":\"Ed, Edd n' Eddy - S01E20 - It's Way Ed.avi\",\"episodeNumber\":\"20\",\"seasonNumber\":\"01\"}" "--duration" "648" "--hash" "8b786dad5cdd63b0" "--size" "82929064"
    [0000000004b47740] [trakt] lua interface error: (call_helper) Command exited with code -1
    [0000000004b47740] [trakt] lua interface error: (call_helper) Command output:
    [0000000004b47740] [trakt] main interface error: connection failed: Connection refused by peer
    [0000000004b47740] [trakt] lua interface: Ed, Edd n' Eddy - S01E20 - It's Way Ed.avi is playing! :) (10.400000/648.000000)
    [0000000004b47740] [trakt] main interface error: connection failed: Connection refused by peer
    [0000000004b47740] [trakt] lua interface error: (call_helper) Command: "--quiet" "resolve" "--meta" "{\"setting\":\" HAS_INDEX\",\"title\":\"Ed, Edd n' Eddy - S01E20\",\"showName\":\"Ed, Edd n' Eddy -\",\"filename\":\"Ed, Edd n' Eddy - S01E20 - It's Way Ed.avi\",\"episodeNumber\":\"20\",\"seasonNumber\":\"01\"}" "--duration" "648" "--hash" "8b786dad5cdd63b0" "--size" "82929064"
    [0000000004b47740] [trakt] lua interface error: (call_helper) Command exited with code -1
    [0000000004b47740] [trakt] lua interface error: (call_helper) Command output:
    [0000000004b47740] [trakt] main interface error: connection failed: Connection refused by peer
    [0000000004b47740] [trakt] lua interface: Ed, Edd n' Eddy - S01E20 - It's Way Ed.avi is playing! :) (19.175000/648.000000)`

Any idea as to what the problem could be?

**Comment by XaF** [comment]
@veselind your issue seems to be different. Could you please open a separate issue so this one is not getting noise ? :)

**Comment by longbaugh01** [comment]
@XaF TraktForVLC is not scrobbling, nor is it showing my current play status on trakt's site.

Is there any other information I can provide you with that would help both of us?

Also, I was using the latest release. I just was mentioning that I tried prior releases to see if that would work. Still with the latest it does not work.

Thanks.

**Comment by XaF** [comment]
@longbaugh01 from the first log, everything seemed to be working, the only thing I see that would justify the video not appearing on trakt.tv is the fact that the video was paused in your log.

```
lua info: The Keepers (2017) - S01E02 - Dream World is paused! :| (6.39e-007/3550.133)
```

And also the current value of the timer (`6.39e-007`) that means you weren't after 30seconds of the beginning. Or using a previous version of TraktForVLC that still had the bug impacting VLC 2.x.

**Comment by longbaugh01** [comment]
@XaF I had to completely uninstall VLC and reinstall the newest version 3.0.2, then reinstall TFV, but now it seems to be working. It's showing what I'm watching on trakt's site and it is scrobbling episodes, BUT as soon as I added the "web" interface and added a lua web password in VLC settings for my Android VLC remote app, it immediately stopped working, and I had to reinstall TFV to get it working again.

So clearly with that issue, and the problems I outlined in my original post (including those installation/re-installation/re-authorization problems) , there are issues that need to be resolved.

That being said, I'm happy that it seems to be working now and I appreciate your hard work and dedication to providing us this way to scrobble to trakt with VLC. Cheers.

**Comment by XaF** [comment]
@longbaugh01 can you tell me if that works properly with your other lua plugin activated? Could you share the name of that other plugin so I can investigate it? Thanks :)

**Comment by longbaugh01** [comment]
@XaF As I said, "BUT as soon as I added the "web" interface and added a lua web password in VLC settings for my Android VLC remote app, it immediately stopped working, and I had to reinstall TFV to get it working again." But I probably should have clarified further and said no, I can't get the app and TFV to work together, which is unfortunate.

The app in question is "VLC Mobile Remote - PC & Mac" https://play.google.com/store/apps/details?id=adarshurs.android.vlcmobileremote&hl=en_US&rdid=adarshurs.android.vlcmobileremote

It has always worked fine until I first started trying to get TFV to work. Hope you figure out what the conflict might be and let me know if you can.
